core/Plugin/Report.php

```diff
@@ -918,21 +918,42 @@ public function getThirdLeveltableDimension()
             return null;
         }
 
-        list($subtableReportModule, $subtableReportAction) = $this->getSubtableApiMethod();
+        return $this->getNthLeveltableDimension($level = 2);
+    }
 
-        $subtableReport = ReportsProvider::factory($subtableReportModule, $subtableReportAction);
-        if (empty($subtableReport) || empty($subtableReport->actionToLoadSubTables)) {
+    /**
+     * Returns the Dimension instance of the subtable report of this report's subtable report based on level.
+     *
+     * @param int $level The subTable level for which dimension is to be determined
+     * @return Dimension|null The subtable report's dimension or null if there is no subtable report or
+     *                        no dimension for the subtable report.
+     * @api
+     */
+    public function getNthLeveltableDimension($level)
```

Comment:
Can we add strict types?
```suggestion
    public function getNthLeveltableDimension(int $level): ?Dimension
```